gpt_bpe.go

```diff
@@ -1123,6 +1151,9 @@ func (encoder *GPTEncoder) Decode(encoded *Tokens) (text string) {
 					bs = append(bs, v...)
 				}
 			}
+			if len(bs) == 0 {
+				continue
+			}
```

Comment:
ah remnant bit, there used to be a bit of code there where it would take out the start or end token if the enclose param was enabled, allowing that part of it to be fully reversable and loss-less between the original text and decode(encode()). I later found out that llama apparently doesn't do this and having a start token in the decoded string is indeed the correct output. Back before i found that out i had a line that was something like if (tokenid == start token id && enclosebos == true) then remove that safetoken. This caused some errors with weird references if it tried to progress with a bs of length 0. Since this isn't needed, will remove.